### PR TITLE
Don't pollute structs when trying to merge invalid keys

### PR DESCRIPTION
**Pull request by mariusbutuc**
As part of [ch1263](https://app.clubhouse.io/planswell/story/1263/clean-up-year-1-values), we want to ensure we can't merge invalid keys into a struct.

Paired on this with @olafura, so @kenips & @boxxxie please have a 👀 .

### CODE REVIEW

File: lib/kitchen_sink/map.ex [hunk 1]
```diff
@@ -57,6 +57,15 @@ defmodule KitchenSink.Map do
     do_deep_merge(left, right)
   end
 
+  defp do_deep_merge(%{__struct__: _} = left, right) do
```
Comment (robertpeacock22) [inline lib/kitchen_sink/map.ex:60]: Comments please 📃

**Review by robertpeacock22 — CHANGES_REQUESTED** [review]

**Review by boxxxie — COMMENTED** [review]
I don't even know if it makes sense to do deep merge with structs

**Comment by mariusbutuc** [comment]
@boxxxie Can you elaborate on this please?

**Comment by boxxxie** [comment]
well, a struct has a lot of restrictions when it is merged

**Review by boxxxie — APPROVED** [review]

**Pull request merged**